examples/TexturedQuadDynamic/main.cpp

```diff
@@ -0,0 +1,99 @@
+// Include Radium base application and its simple Gui
+#include <Gui/BaseApplication.hpp>
+#include <Gui/RadiumWindow/SimpleWindowFactory.hpp>
+
+// include the Engine/entity/component interface
+#include <Core/Asset/BlinnPhongMaterialData.hpp>
+#include <Core/Geometry/MeshPrimitives.hpp>
+#include <Core/Tasks/Task.hpp>
+#include <Engine/Data/TextureManager.hpp>
+#include <Engine/FrameInfo.hpp>
+#include <Engine/Scene/EntityManager.hpp>
+#include <Engine/Scene/GeometryComponent.hpp>
+#include <Engine/Scene/GeometrySystem.hpp>
+
+#include <QTimer>
+
+/// Compute a texture wrt to a given timestamp
+class DynamicTextureSystem : public Ra::Engine::Scene::System
+{
+    unsigned char data[192 * 512];
+    const std::string textureName { "myTexture" };
+    Ra::Engine::Data::TextureManager* m_texManager { nullptr };
+
+    inline void computeTexture( Scalar time ) {
+        // fill with some function
+        for ( int i = 0; i < 192; ++i ) {
+            for ( int j = 0; j < 512; j++ ) {
+                data[( i * 512 + j )] =
+                    (unsigned char)( 255.0 * std::abs( std::sin( j * i * M_PI / 64.0 ) *
+                                                       std::cos( j * i * M_PI / 96.0 ) ) );
+            }
+        }
+    }
+
+  public:
+    inline DynamicTextureSystem( Ra::Engine::Data::TextureManager* mng ) : m_texManager( mng ) {
+        assert( m_texManager != nullptr );
+
+        /// Create texture
+        this->computeTexture( 0 );
+        auto& myTexture = m_texManager->addTexture( textureName, 192, 512, data );
+        // these values will be used when engine initialize texture GL representation.
+        myTexture.format         = gl::GLenum::GL_RED;
+        myTexture.internalFormat = gl::GLenum::GL_R8;
+        m_texManager->loadTexture( myTexture, false );
```

Comment:
Use getOrLoadTexture here

Reply:
If I do so, the texture is not accessible in the shader anymore (quad is black)

Reply:
And if I use `LoadTexture` I crash at the CORE_ASSERT when calling `updateTextureContent` in the task.